src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporterCommandLineOptionsProvider.cs

```diff
@@ -45,12 +56,26 @@ public Task<ValidationResult> ValidateOptionArgumentsAsync(CommandLineOption com
         {
             NoProgressOption => ValidationResult.ValidTask,
             NoAnsiOption => ValidationResult.ValidTask,
+            AnsiOption => IsValidAnsiArgument(arguments[0])
+                ? ValidationResult.ValidTask
+                : ValidationResult.InvalidTask(PlatformResources.TerminalAnsiOptionInvalidArgument),
             OutputOption => OutputOptionNormalArgument.Equals(arguments[0], StringComparison.OrdinalIgnoreCase) || OutputOptionDetailedArgument.Equals(arguments[0], StringComparison.OrdinalIgnoreCase)
                 ? ValidationResult.ValidTask
                 : ValidationResult.InvalidTask(PlatformResources.TerminalOutputOptionInvalidArgument),
             _ => throw ApplicationStateGuard.Unreachable(),
         };
 
+    private static bool IsValidAnsiArgument(string argument)
+        => AnsiOptionAutoArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionOnArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionTrueArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionEnableArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOption1Argument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionOffArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionFalseArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOptionDisableArgument.Equals(argument, StringComparison.OrdinalIgnoreCase)
+            || AnsiOption0Argument.Equals(argument, StringComparison.OrdinalIgnoreCase);
```

Comment:
this will be hard to replicate for any similar option that needs on / off , and potentially auto. Can you make this more reusable. 

Feels like CommandLineOption type we are creating should be BoolAutoCommandLineOption, and have this built-in. 

But that does not fit the design here, so at least a reusable method that will validate this.